analyses/cell-type-ewings/exploratory_analysis/08-merged-celltypes.Rmd

```diff
@@ -0,0 +1,694 @@
+  )
+
+all_info_df |> 
+  dplyr::count(final_annotation) |> 
+  dplyr::arrange(desc(n))
+```
+
+Let's remake the density plot to see if we have pulled out the tumor cells from the normal cell types like muscle cells. 
+
+```{r}
+cluster_density_plot(all_info_df, genesets_of_interest, "final_lumped", "AUC")
+```
+
+It looks like generally the `tumor EWS-high` cells show high expression of both upregulated gene sets and `tumor EWS-low` cells show high expression of both downregulated gene sets, so I feel good about this! 
+
+I will also note that there are some muscle cells that show slight expression of `aynaud-ews-targets`, but show little to no expression of `staege` targets. 
+I'm still a little worried these are actually tumor cells, but because they only show expression of one of the genesets, we are probably okay to keep them as is noting that the expression profile of muscle cells is very similar to those of tumor cells, this being sarcoma. 
+
+## Identifying proliferative cells
+
+The last tumor cell state that we want to identify is if any cells can be considered proliferative using the set of proliferative markers in `tumor-cell-state-markers.tsv`. 
+Let's look at the mean expression of these markers across all cells. 
+Since we are plotting the proliferative markers, let's also look at our other custom marker gene sets. 
+
+```{r}
+# get columns with mean expression of custom gene sets 
+mean_exp_columns <- colnames(all_info_df)[which(endsWith(colnames(all_info_df), "_mean"))]
+# make sure order is defined by cell type order
+all_info_df <- all_info_df |> 
+  dplyr::arrange(final_lumped)
+
+single_annotation_heatmap(
+  all_info_df, 
+  exp_columns = mean_exp_columns, 
+  cell_type_column = "final_lumped", 
+  legend_title = "AUC"
+)
+```
+
+It looks like we do have a set of cells that are `tumor EWS-high` and also have high expression of proliferative markers. 
+Because of this, let's label any cells that are `tumor EWS-high` and have mean expression of proliferative markers > 0 as `tumor EWS-high proliferative`. 
+
+```{r}
+# set desired order for plotting final annotations later
+cell_type_order <- c(
+  "tumor EWS-high",
+  "tumor EWS-high proliferative",
+  "tumor EWS-low",
+  "muscle cell", 
+  "smooth muscle cell",
+  "fibroblast",
+  "endothelial cell",
+  "macrophage",
+  "mature T cell",
+  "Unknown",
+  "All remaining cell types"
+)
+
+all_info_df <- all_info_df |> 
+  dplyr::mutate(
+    final_annotation = dplyr::if_else(
+      final_annotation == "tumor EWS-high" & proliferative_mean > 0,
+      "tumor EWS-high proliferative",
+      final_annotation
+    ), 
+    final_lumped = final_annotation |>
+      forcats::fct_lump_n(10, other_level = "All remaining cell types", ties.method = "first") |>
+      forcats::fct_infreq() |>
+      forcats::fct_relevel(cell_type_order)
+  )
+
+all_info_df |> 
+  dplyr::count(final_annotation) |>
+  dplyr::arrange(desc(n))
+```
```

Comment:
Can we also look at a table of how many libraries each cell type appears in? With the smaller cell types, we're not going to see them even close to uniformly distributed. I think this might be informative, like if a cell type is only present in 1 library, it could be more likely to be mislabeled...but it could also certainly be real  🤷‍♀️